### DISCUSSION TITLE
How to convert netcdf file to dfs3 file?

### DISCUSSION THREAD

**Discussion by Junik4** [DHI/mikeio#688] [Q&A] [answered]
I have nc data from HYCOM.
I want to input this data to Mike3. 
but I feel difficult to convert this data to .dfs2 for boundary conditions and .dfs3 for initial conditions.
because the example in mikeio is from ECMWF data.
can you help me, the script python for convert netcdf file to dfs2 and dfs3 file?
thanks

**Comment by ecomodeller** [comment]
An easy way to get boundary conditions for MIKE 3 is to use [MIKE Data Link](https://www.mikepoweredbydhi.com/products/mike-cloud/mike-data-link/).

**Reply by carolinelai777** [reply]
Is Datalink still working? I tried last week and failed to extract anything :(

**Reply by ecomodeller** [reply]
@nielsriegels any comment?

**Comment by carolinelai777** [comment]
Is there a solution in year 2026? I am looking for a script to prepare initial condition in dfs3 format from HYCOM and Copernicus.

**Reply by ecomodeller** [reply]
FYI @JesperGr

**Reply by JesperGr** [reply]
You may try take a look at this dfs2 example:

https://github.com/DHI/mikecore-python/blob/5edcf6a33156ace5869f854429e6f700a9247fd0/tests/examples_dfs2.py#L149

The difference between dfs3 and dfs2 is setting of the spatial axis: It must use `factory.CreateAxisEqD3`, and also data has to be 3D in the `file.WriteItemTimeStep`. Custom block and static items can be ignored.

Assuming data is a in the form of equidistant 3D grid data set, then this approach can be used to imported data directly into a dfs3 file

**Reply by carolinelai777** [reply]
Thanks Henrik and Jesper!! I will look at this tomorrow. 
PS: I am too excited now.

**Reply by carolinelai777** [reply]
I used this example to post process BRAN data in Australia. It works well. But unfortunately both HYCOM and Copernicus are non equidistant... I cannot write them to dfs2 or dfs3 directly.

**Reply by carolinelai777** [reply, marked as answer]
Thanks to Mohammad (mohm), I am downloading and processing HYCOM into 3D dfsu using hycompy. 
Once again, the day is saved!! :D
It's a very nice tool. We should make this more robust and share with others.

**Reply by Crissan85** [reply]
@carolinelai777 it is possible to get a example? Please 🙏

**Reply by carolinelai777** [reply]
You can get it from Github, search for hycompy: 
<img width="1564" height="902" alt="image" src="https://github.com/user-attachments/assets/bcfe8cf5-6b0e-493f-a25c-00c391b2306e" />

**Reply by carolinelai777** [reply]
I got help from Mohammad to get this on my laptop. Please contact him if you need any help ;)

**Reply by carolinelai777** [reply]
I also write Copernicus into 3D dfsu because I need the salinity and temperature in 3D map as my initial condition.

**Reply by Crissan85** [reply]
thanks @carolinelai777 , I was always looking for how to generate the same types of 3D initial conditions 🙌

**Reply by Crissan85** [reply]
Hi @mohm-dhi , I'd like to ask if it's possible to get help generating these initial conditions in 3D (DFS3 or DFSU) using Hycompy. Thank you in advance.

**Reply by mohm-dhi** [reply]
Yes, you can use hycompy to generate initial conditions (dfsu) and boundary conditions as well. The code needs some improvements to make it easier for anyone to use it, which I am hoping a project budget support that so I can charge my hours to it. It needs 1-2 days of work.

**Reply by Crissan85** [reply]
thanks @mohm-dhi , I currently don't have access to hycompy. Is it possible to gain access? It's fine if it needs improvements; I can make some in the meantime, and it can be shared.
Thanks in advance.

**Reply by mohm-dhi** [reply]
> thanks @mohm-dhi , I currently don't have access to hycompy. Is it possible to gain access? It's fine if it needs improvements; I can make some in the meantime, and it can be shared. Thanks in advance.

the repository is private. Sorry I do not recognize you from your profile and username, are you working for DHI? if yes, you can have access to the private repositories by contacting hotline I believe or one of the DHI GitHub admins.

**Reply by Crissan85** [reply]
> > thanks @mohm-dhi , I currently don't have access to hycompy. Is it possible to gain access? It's fine if it needs improvements; I can make some in the meantime, and it can be shared. Thanks in advance.
> 
> the repository is private. Sorry I do not recognize you from your profile and username, are you working for DHI? if yes, you can have access to the private repositories by contacting hotline I believe or one of the DHI GitHub admins.

No, I'm just a regular user. I understand that's not possible. Anyway, thank you.
